test/stage2/compare_output.zig

```diff
@@ -31,6 +31,11 @@ pub fn addCases(ctx: *TestContext) !void {
             \\export fn _start() noreturn {
             \\    print();
             \\
+            \\    const a: u32 = 2;
+            \\    const b: ?u32 = a;
+            \\    const c = b.?;
+            \\    if (c != 2) unreachable;
+            \\
```

Comment:
can you leave the "regular old hello world" update alone and add a new `addCompareOutput` below with this test case?

You should also be able to test your new compile error by using `case.addError`.

I think the plan forward is going to be to unify the compare_output.zig and compile_errors.zig test case files and then split them up by feature, rather than by nature of the test.